desktop-widgets/tab-widgets/TabDiveInformation.cpp

```diff
@@ -93,9 +93,12 @@ void TabDiveInformation::closeWarning()
 }
 
 void TabDiveInformation::updateWaterTypeWidget()
-{    // Decide on whether to show the water type/salinity combobox or not
-	if (prefs.salinityEditDefault || manualDive)
-	{         // if the preference setting has been checked or this is a manually-entered dive
+{
+    struct divecomputer *currentdc = get_dive_dc(current_dive, dc_number);
+    bool hasDCSalinity = currentdc && currentdc->salinity != 0;
+    // Decide on whether to show the water type/salinity combobox or not
+	if (prefs.salinityEditDefault || manualDive || !hasDCSalinity)
```

Comment:
I think you can simplify this by getting rid of `manualDive` here (and replace it with `!hasDcSalinity` in the other places where it is used), as `hasDcSalinity` will evaluate to false for manual dives.

Reply:
>as hasDcSalinity will evaluate to false for manual dives.

Why? Because manual dives doesnt have "dc" or beacuse its doesnt have salinity? Or both?

Reply:
Done! https://github.com/subsurface/subsurface/pull/3854/commits/bad184cb0dd640e0d68f0c95aa021dfbb996efb9